core/drivers/pm/sam/at91_pm.c

```diff
@@ -311,13 +325,27 @@ TEE_Result atmel_pm_suspend(uintptr_t entry, struct sm_nsec_ctx *nsec)
 
 	sm_save_unbanked_regs(&nsec->ub_regs);
 
+	/*
+	 * In order to run code for low-power out of SRAM without abort,
+	 * configure regions with write permission with not forced to
+	 * XN (Execute-never) attribute.
+	 */
+	if (read_id_pfr1() & IDPFR1_VIRT_MASK) {
```

Comment:
Wouldn't it make sense to read `SCTLR` and if `SCTLR_WXN` is set clear it, regardless of `ID_PFR1`?
The bit is reserved as `0` if not supported.